sdcflows/cli/parser.py

```diff
@@ -0,0 +1,356 @@
+        def __call__(self, parser, namespace, values, option_string=None):
+            setattr(namespace, self.dest, _parse_participant_labels(" ".join(values)))
+
+    def _path_exists(path, parser):
+        """Ensure a given path exists."""
+        if path is None or not Path(path).exists():
+            raise parser.error(f"Path does not exist: <{path}>.")
+        return Path(path).expanduser().absolute()
+
+    def _min_one(value, parser):
+        """Ensure an argument is not lower than 1."""
+        value = int(value)
+        if value < 1:
+            raise parser.error("Argument can't be less than one.")
+        return value
+
+    def _to_gb(value):
+        scale = {"G": 1, "T": 10**3, "M": 1e-3, "K": 1e-6, "B": 1e-9}
+        digits = "".join([c for c in value if c.isdigit()])
+        n_digits = len(digits)
+        units = value[n_digits:] or "G"
+        return int(digits) * scale[units[0]]
+
+    def _bids_filter(value):
+        from json import loads
+
+        if value and Path(value).exists():
+            return loads(Path(value).read_text())
+
+    parser = ArgumentParser(
+        description=f"""\
+SDCFlows {config.environment.version}
+
+Estimate fieldmaps available in a BIDS-compliant MRI dataset.""",
+        formatter_class=ArgumentDefaultsHelpFormatter,
+    )
+    PathExists = partial(_path_exists, parser=parser)
+    PositiveInt = partial(_min_one, parser=parser)
+
+    parser.add_argument(
+        "bids_dir",
+        action="store",
+        type=PathExists,
+        help="The root folder of a BIDS valid dataset (sub-XXXXX folders should "
+        "be found at the top level in this folder).",
+    )
+    parser.add_argument(
+        "output_dir",
+        action="store",
+        type=Path,
+        help="The directory where the output files "
+        "should be stored. If you are running group level analysis "
+        "this folder should be prepopulated with the results of the "
+        "participant level analysis.",
+    )
+    parser.add_argument(
+        "analysis_level",
+        action="store",
+        nargs="+",
+        help="Level of the analysis that will be performed. "
+        "Multiple participant level analyses can be run independently "
+        "(in parallel) using the same output_dir.",
+        choices=["participant", "group"],
+    )
+
+    # optional arguments
+    parser.add_argument(
+        "--version", action="version", version=f"SDCFlows {config.environment.version}"
+    )
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        dest="verbose_count",
+        action="count",
+        default=0,
+        help="Increases log verbosity for each occurrence, debug level is -vvv.",
+    )
+
+    # main options
+    g_bids = parser.add_argument_group("Options related to BIDS")
+    g_bids.add_argument(
+        "--participant-label",
+        "--participant_label",
+        "--participant-labels",
+        "--participant_labels",
+        dest="participant_label",
```

Comment:
I would be inclined not to have synonyms when they aren't required for backwards compatibility.

```suggestion
        "--participant-label",
```